framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -203,43 +223,63 @@ private static String formatArrayType(ATypeElement e, String arrayType) {
      *
      * @param e same as above, but can become null if scene-lib did not fill in the inner types,
      *     which happens when they do not have annotations
-     * @param arrayType same as above
+     * @param arrayTypes the array parts of the array type (i.e. the parts after the component
+     *     type). Must contain at least one '['.
+     * @param componentType the component type of the array
      * @param result the string builder containing the array types seen so far
-     * @return the formatted string, as above, with a trailing space
+     * @return the formatted string, without a trailing space
      */
     private static String formatArrayTypeImpl(
-            @Nullable ATypeElement e, String arrayType, StringBuilder result) {
-        String nextComponentType =
-                arrayType.indexOf('[') == -1
-                        ? null
-                        : arrayType.substring(0, arrayType.lastIndexOf('['));
-        // base case when the component is a non-array type
-        if (nextComponentType == null) {
-            String componentAsString = arrayType + " " + result.toString();
-            if (e != null) {
-                return formatAnnotations(e.tlAnnotationsHere) + componentAsString;
-            } else {
-                return componentAsString;
-            }
+            @Nullable ATypeElement e,
+            String arrayTypes,
+            String componentType,
+            StringBuilder result) {
+        // print the next type:
```

Comment:
Is this the next array level, or something else?
Also, it doesn't seem to be printing, but appending or building up a string.